core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -186,12 +205,54 @@ public void start() {
                     log.warn("Exception while pulling images, using local images if available", e);
                 }
             }
+            resolveDockerComposeVersion();
+            registerServices();
             createServices();
             startAmbassadorContainers();
             waitUntilServiceStarted();
         }
     }
 
+    private void resolveDockerComposeVersion() {
+        if (this.localCompose) {
+            this.composeV2 = LocalDockerCompose.IS_COMPOSE_V2;
+            return;
+        }
+        this.composeV2 = TestcontainersConfiguration.getInstance().isComposeV2Enabled();
+    }
+
+    private void registerServices() {
+        this.exposedServices.forEach(exposedService -> {
```

Comment:
out of curiosity: why not good old `for-in`? :D

Reply:
since Java 8 is the first thing I used by default unless for loop is really needed

Reply:
generally we stick to `for-in`. It takes the same amount of code, but has some benefits (better stacktraces, can use `break`, etc etc)